web/src/stream_create_subscribers_data.ts

```diff
@@ -1,43 +1,46 @@
 import {page_params} from "./page_params";
 import * as people from "./people";
+import type {User} from "./people";
 
-let user_id_set;
+let user_id_set: Set<number | undefined>;
 
-export function initialize_with_current_user() {
+export function initialize_with_current_user(): void {
     const current_user_id = page_params.user_id;
-    user_id_set = new Set();
+    user_id_set = new Set<number | undefined>();
     user_id_set.add(current_user_id);
```

Comment:
We can omit generic type for `Set` here since we have already specified it above when declaring `user_id_set`.

Also I think we should conditionally add `current_user_id` to `user_id_set` to avoid `undefined` type.
```ts
export function initialize_with_current_user(): void {
    const current_user_id = page_params.user_id;
    user_id_set = new Set();
    if (current_user_id !== undefined) {
        user_id_set.add(current_user_id);
    }
}
```

Reply:
What I did is that I traced all the way back to verify `page_params.user_id` will not be `undefined` at the point that the function is being called. This is because when user is able to create and initialize a stream, this user is authenticated and thus a `user_id` is always available. Therefore, adding an assertion when undefined user id is detected to handle a confirmed error.

Reply:
Yeah an assert statement sounds good 👍